### ISSUE TITLE
The PICSA crops dialogue 

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5388]
I have now tried testing the crops dialogue.
One small cosmetic point is that the last control (Season End) is not remembered.  It is cleared every time.

I have been trying with the Tamale data.  The file is in the Climatic Documentation folder (Data sub-folder) and also here:
[Tamale_for_crops.zip](https://github.com/africanmathsinitiative/R-Instat/files/3312679/Tamale_for_crops.zip)

In the dialogue I used start_doy and also End_season2.

The table was for starting dates: 75, 95, 115
Rain total 300, 500, 700
Length of crop, 80, 100, 120
Note the earliest starting date I used is before it ever started - so should give zeros.

a) I first unticked the Require planting before starting.  So that (to me) is all 3 conditions.  That's when |I should get zeros for the day 75 planting - but I didn't.  
b) Then I ticked the require planting before zeros.  That should give bigger values, because it is just the 2 conditions, of length and amount.  It gives all zeros.

I hope it is a simple setting to change perhaps?

**Comment by dannyparsons** [comment]
There is an issue with the overall condition column not being correct, which I've identified and fixing.

On the checkbox:
- When the "Require start day before planting day" checkbox is **checked** then there are 3 conditions. The start day in the data must be before the specified planting day.
- When the "Require start day before planting day" checkbox is **unchecked** then there are 2 conditions. There is no check done on the start day (the planting day specified is still used to calculate length and totals)

I think that's the opposite to what you expected? To me, how it is currently makes sense, but we could reword if its confusing?

**Comment by dannyparsons** [comment]
Also, notice that in the `crop_def` data frame, when the checkbox is checked there are 4 logical columns (3 conditions and overall), when it is unchecked there are 3 (2 conditions and overall). These can be used for verification of the calculations.

I did notice in the Tamale data one year where the start day was 75, so there are non zero probabilities for a planting day 75 as the checks are not using strict inequalities i.e. `start_day <= planting day` is the condition. I'm not sure we discussed whether to use `<` or `<=` for these, but can easily be changed.
